### PR TITLE
Tweak OIDC middleware to support non-docker workflow

### PR DESCRIPTION
**Pull request by rehandalal**
cc: @tiftran - since you've asked me before about setting up without docker this might be of interest

r?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -70,6 +70,7 @@ target/
 
 # Environment file
 .env
+.envrc
```
Comment (rehandalal) [inline .gitignore:73]: added to support [direnv](https://direnv.net/)

File: app/experimenter/openidc/middleware.py [hunk 2]
```diff
@@ -30,7 +30,8 @@ def __call__(self, request):
         except Resolver404:
             pass
 
-        openidc_email = request.META.get(settings.OPENIDC_EMAIL_HEADER, None)
+        default_email = settings.DEV_USER_EMAIL if settings.DEBUG else None
```
Comment (jaredlockhart) [inline app/experimenter/openidc/middleware.py:33]: Coulda done the debug check in settings.py and set DEV_USER_EMAIL to None in there and then just refer to here and use the None check on the following line to catch both.

Comment (jaredlockhart) [reply app/experimenter/openidc/middleware.py:33]: But that's just a thought, this is fine as is.

**Comment by rehandalal** [comment]
> Doesn't this need the settings.py change as well?

It did when I had added a new setting but then I realized we already had one with an email and this just made the behavior consistent with Docker where you cannot choose the dev email (afaict)

**Pull request merged**